src/main/cpp/states/ballhopper/BallHopperStateMgr.cpp

```diff
@@ -136,6 +137,19 @@ BallHopperStateMgr::BallHopperStateMgr() : m_currentState(nullptr),
 /// @return void
 void BallHopperStateMgr::RunCurrentState()
 {
+    //process teleop/manual interrupts
```

Comment:
Ball hopper states should be driven by the shooter states

Reply:
These changes were being used for testing the shooter.  I don't think the ballhopper was tied to the shooter buttons so this was just a bandaid fix while we were there and we will have to make sure the ballhopper is being controlled by the shooter states.

Reply:
This block should have caused the hopper to go into hold state; wasn't it working?      

  if ( controller->IsButtonPressed( TeleopControl::FUNCTION_IDENTIFIER::SHOOTER_PREPARE_TO_SHOOT ))
        {
            Logger::GetLogger()->ToNtTable(ntName, "Current State", "Prepare to Shoot");
            SetCurrentState( SHOOTER_STATE::GET_READY, false );
            BallHopperStateMgr::GetInstance()->SetCurrentState( BallHopperStateMgr::HOLD, false);
            BallTransferStateMgr::GetInstance()->SetCurrentState( BallTransferStateMgr::BALL_TRANSFER_STATE::TO_SHOOTER, false );
        }

Similarly, the particular color shooting spots right now indicate rapid release and probably should be slow release.

Reply:
I wasn't the one controlling and testing the shooter but I don't think they were pressing the prepare to shoot button.  From the testing yesterday, I'm not sure if they want us to use the slow release function.  It sounded like they wanted to release all the balls at once for better accuracy.  I can figure out what's going on tonight and sort out this pull request.

Reply:
ok, sounds good